方法返回InputStream时expected map/object at java.lang.String异常

dubbo版本：3.2.14
hessian版本：3.2.13

示例代码
* NatureReserveSliceService.java
```java
public InputStream test() {
    System.out.println("被调用了 返回 InputStream");
    return new ByteArrayInputStream(new byte[]{21});
}
```
* 调用
```java
InputStream test = sliceService.test();
```

* 错误信息
```
Caused by: org.apache.dubbo.remoting.RemotingException: com.alibaba.com.caucho.hessian.io.HessianProtocolException: expected map/object at java.lang.String
com.alibaba.com.caucho.hessian.io.HessianProtocolException: expected map/object at java.lang.String
	at com.alibaba.com.caucho.hessian.io.AbstractDeserializer.error(AbstractDeserializer.java:131)
	at com.alibaba.com.caucho.hessian.io.AbstractMapDeserializer.readObject(AbstractMapDeserializer.java:70)
	at com.alibaba.com.caucho.hessian.io.Hessian2Input.readObject(Hessian2Input.java:2297)
	at com.alibaba.com.caucho.hessian.io.Hessian2Input.readObject(Hessian2Input.java:2104)
	at org.apache.dubbo.common.serialize.hessian2.Hessian2ObjectInput.readObject(Hessian2ObjectInput.java:117)
	at org.apache.dubbo.common.serialize.ObjectInput.readAttachments(ObjectInput.java:85)
	at org.apache.dubbo.common.serialize.DefaultSerializationExceptionWrapper$ProxyObjectInput.readAttachments(DefaultSerializationExceptionWrapper.java:197)
	at org.apache.dubbo.rpc.protocol.dubbo.DecodeableRpcResult.handleAttachment(DecodeableRpcResult.java:194)
	at org.apache.dubbo.rpc.protocol.dubbo.DecodeableRpcResult.decode(DecodeableRpcResult.java:111)
	at org.apache.dubbo.rpc.protocol.dubbo.DecodeableRpcResult.decode(DecodeableRpcResult.java:153)
	at org.apache.dubbo.remoting.transport.DecodeHandler.decode(DecodeHandler.java:61)
	at org.apache.dubbo.remoting.transport.DecodeHandler.received(DecodeHandler.java:49)
	at org.apache.dubbo.remoting.transport.dispatcher.ChannelEventRunnable.run(ChannelEventRunnable.java:64)
	at org.apache.dubbo.common.threadpool.ThreadlessExecutor$RunnableWrapper.run(ThreadlessExecutor.java:151)
	at org.apache.dubbo.common.threadpool.ThreadlessExecutor.waitAndDrain(ThreadlessExecutor.java:77)
	at org.apache.dubbo.rpc.AsyncRpcResult.get(AsyncRpcResult.java:219)
	at org.apache.dubbo.rpc.protocol.AbstractInvoker.waitForResultIfSync(AbstractInvoker.java:292)
	at org.apache.dubbo.rpc.protocol.AbstractInvoker.invoke(AbstractInvoker.java:194)
	at org.apache.dubbo.rpc.listener.ListenerInvokerWrapper.invoke(ListenerInvokerWrapper.java:71)
	at org.apache.dubbo.rpc.filter.RpcExceptionFilter.invoke(RpcExceptionFilter.java:40)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:349)
	at com.hdi.bhd.dubbo.filter.CallServiceFilter.invoke(CallServiceFilter.java:35)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:349)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CallbackRegistrationInvoker.invoke(FilterChainBuilder.java:197)
	at org.apache.dubbo.rpc.protocol.ReferenceCountInvokerWrapper.invoke(ReferenceCountInvokerWrapper.java:106)
	at org.apache.dubbo.rpc.cluster.support.AbstractClusterInvoker.invokeWithContext(AbstractClusterInvoker.java:412)
	at org.apache.dubbo.rpc.cluster.support.FailoverClusterInvoker.doInvoke(FailoverClusterInvoker.java:82)
	at org.apache.dubbo.rpc.cluster.support.AbstractClusterInvoker.invoke(AbstractClusterInvoker.java:366)
	at org.apache.dubbo.rpc.cluster.router.RouterSnapshotFilter.invoke(RouterSnapshotFilter.java:46)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:349)
	at org.apache.dubbo.monitor.support.MonitorFilter.invoke(MonitorFilter.java:108)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:349)
	at org.apache.dubbo.rpc.cluster.filter.support.MetricsClusterFilter.invoke(MetricsClusterFilter.java:57)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:349)
	at org.apache.dubbo.rpc.protocol.dubbo.filter.FutureFilter.invoke(FutureFilter.java:52)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:349)
	at org.apache.dubbo.rpc.cluster.filter.support.ObservationSenderFilter.invoke(ObservationSenderFilter.java:62)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:349)
....

```

## 经过debug后大致确定问题
1. 代码执行到 `org.apache.dubbo.rpc.protocol.dubbo.DecodeableRpcResult#decode(org.apache.dubbo.remoting.Channel, java.io.InputStream)` 109行
```
            case DubboCodec.RESPONSE_VALUE_WITH_ATTACHMENTS:
                handleValue(in); // 读取返回值，进入次方法
                handleAttachment(in);  
                break;
```
2. 执行到`com.alibaba.com.caucho.hessian.io.Hessian2Input#readObject(java.lang.Class, java.lang.Class<?>...)` 2297 行 获取`InputStreamDeserializer`返序列化示例，并调用`readObject`方法
```
Object value = findSerializerFactory().getDeserializer(expectedClass).readObject(this);
```
4. 在`InputStreamDeserializer.readObject`调用了`in.readInputStream()`
5. 在`com.alibaba.com.caucho.hessian.io.Hessian2Input#readInputStream`方法中读取到的tag值为65，执行`case 'b':` ，问题可能就在这，在这里只是读取了数据长度，并没有读取数据就结束了
```
case 'b': //maybe it's a mistype of BC_BINARY_CHUNK
    _isLastChunk = tag == 'B';
    _chunkLength = (read() << 8) + read(); // 获取数据长度
    break; // 结束 switch
```
7. 代码执行到`com.alibaba.com.caucho.hessian.io.Hessian2Input#readInputStream`方法3424行时只返回了`ReadInputStream`实例，也没用读取数据就返回了
8. 执行到`org.apache.dubbo.rpc.protocol.dubbo.DecodeableRpcResult#decode(org.apache.dubbo.remoting.Channel, java.io.InputStream)` 111行`handleAttachment`方法时，因为前面的数据没有读取，参数`in`中的`_offset`值没有变化，所以导致`handleAttachment`方法读取失败。

* 以上是debug得到的信息，个人觉得传输`InputStream`时应该把数据放到末尾，先读取`Attachment`再读取`InputStream`，因为可能`InputStream`比较大

> Can you pls test on Dubbo 3.3 using hessian-lite 4.x?

I upgraded the version
dubbo version: 3.3.1
hessian-lite version: 4.0.3

A new error has occurred:
```
Caused by: java.lang.IllegalArgumentException: [Serialization Security] Serialized class java.io.InputStream has not implement Serializable interface. Current mode is strict check, will disallow to deserialize it by default. 
	at org.apache.dubbo.common.utils.DefaultSerializeClassChecker.loadClass(DefaultSerializeClassChecker.java:119) ~[dubbo-3.3.1.jar:3.3.1]
	at org.apache.dubbo.common.serialize.hessian2.Hessian2SerializerFactory.getDefaultDeserializer(Hessian2SerializerFactory.java:68) ~[dubbo-3.3.1.jar:3.3.1]
	at com.alibaba.com.caucho.hessian.io.SerializerFactory.loadDeserializer(SerializerFactory.java:545) ~[hessian-lite-4.0.3.jar:4.0.3]
	at com.alibaba.com.caucho.hessian.io.SerializerFactory.getDeserializer(SerializerFactory.java:474) ~[hessian-lite-4.0.3.jar:4.0.3]
	at com.alibaba.com.caucho.hessian.io.Hessian2Input.readObject(Hessian2Input.java:2408) ~[hessian-lite-4.0.3.jar:4.0.3]
	at com.alibaba.com.caucho.hessian.io.Hessian2Input.readObject(Hessian2Input.java:2218) ~[hessian-lite-4.0.3.jar:4.0.3]
	at org.apache.dubbo.common.serialize.hessian2.Hessian2ObjectInput.readObject(Hessian2ObjectInput.java:118) ~[dubbo-3.3.1.jar:3.3.1]
	at org.apache.dubbo.common.serialize.hessian2.Hessian2ObjectInput.readObject(Hessian2ObjectInput.java:129) ~[dubbo-3.3.1.jar:3.3.1]
	at org.apache.dubbo.common.serialize.DefaultSerializationExceptionWrapper$ProxyObjectInput.readObject(DefaultSerializationExceptionWrapper.java:170) ~[dubbo-3.3.1.jar:3.3.1]
	... 10 common frames omitted
```

* And I haven't found how to customize InputStream Serializable

--- a/dubbo-serialization/dubbo-serialization-hessian2/src/main/java/org/apache/dubbo/common/serialize/hessian2/Hessian2SerializerFactory.java
+++ b/dubbo-serialization/dubbo-serialization-hessian2/src/main/java/org/apache/dubbo/common/serialize/hessian2/Hessian2SerializerFactory.java
@@ -67,6 +67,10 @@ public class Hessian2SerializerFactory extends SerializerFactory {
 
     @Override
     protected Deserializer getDefaultDeserializer(Class cl) {
+        if (InputStream.class.equals(cl)) {
+            return InputStreamDeserializer.DESER;
+        }
+
         try {
             // pre-check if class is allow
             defaultSerializeClassChecker.loadClass(getClassLoader(), cl.getName());
@@ -76,9 +80,7 @@ public class Hessian2SerializerFactory extends SerializerFactory {
 
         checkSerializable(cl);
 
-        if (InputStream.class.equals(cl)) {
-            return InputStreamDeserializer.DESER;
-        } else if (RecordUtil.isRecord(cl)) {
+        if (RecordUtil.isRecord(cl)) {
             return new RecordDeserializer(cl, getFieldDeserializerFactory());
         } else {
             if (isEnableUnsafeSerializer()) {

